src/uct/ib/mlx5/ib_mlx5.h

```diff
@@ -112,29 +131,20 @@ struct mlx5_cqe64* uct_ib_mlx5_check_completion(uct_ib_mlx5_cq_t *cq,
                                                 struct mlx5_cqe64 *cqe);
 
 
-static inline void uct_ib_mlx5_wqe_set_data_seg(struct mlx5_wqe_data_seg *seg,
-                                                void *ptr, unsigned length,
-                                                uint32_t lkey)
-{
-    seg->byte_count = htonl(length);
-    seg->lkey       = htonl(lkey);
-    seg->addr       = htonll((uintptr_t)ptr);
-}
-
 static inline struct mlx5_cqe64* uct_ib_mlx5_get_cqe(uct_ib_mlx5_cq_t *cq,
-                                                     size_t cqe_size)
+                                                     int cqe_size_log)
 {
     struct mlx5_cqe64 *cqe;
     unsigned index;
     uint8_t op_own;
 
     index  = cq->cq_ci;
-    cqe    = cq->cq_buf + (index & (cq->cq_length - 1)) * cqe_size;
+    cqe    = cq->cq_buf + ((index & (cq->cq_length - 1)) << cqe_size_log);
     op_own = cqe->op_own;
 
-    if ((op_own & MLX5_CQE_OWNER_MASK) == !(index & cq->cq_length)) {
+    if (ucs_likely((op_own & MLX5_CQE_OWNER_MASK) == !(index & cq->cq_length))) {
```

Comment:
i think this should be "unlikely" since we want to optimize the case when there is completion